tracer/tracer.go

```diff
@@ -0,0 +1,166 @@
+package tracer
+
+// TODO: change package
+
+import (
+	"time"
+
+	opentracing "github.com/opentracing/opentracing-go"
+)
+
+// Tracer extends the opentracing.Tracer interface with methods to
+// probe implementation state, for use by tracer consumers.
+type Tracer interface {
+	opentracing.Tracer
+	Options() Options
+}
+
+type SpanRecorder interface {
```

Comment:
Can we just rename this to a `SpanReporter` interface? And call our implementation `WavefrontSpanReporter`?

Recorder conveys a different idea than reporting.